File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/paddock/PaddockToSellListMessage.scala





















// Generated on 05/03/2013 10:34:45
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait PaddockToSellListMessage extends MessageDeserializer {
  val messageId = 6138

  protected trait Message extends BaseMessage with Serializable {
    val messageId = PaddockToSellListMessage.messageId

    val pageIndex: Short
    val totalPage: Short
    val paddockList: Seq[PaddockInformationsForSell.TypeImpl]

    def serialize(buf: Buffer) {
      buf.writeShort(pageIndex)
      buf.writeShort(totalPage)
      buf.writeUShort(paddockList.length)
      for (entry <- paddockList) {
        entry.serialize(buf)
      }
    }

  }

  protected def create(pageIndex: Short, totalPage: Short, paddockList: Seq[PaddockInformationsForSell.TypeImpl]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val pageIndex = buf.readShort()
    val totalPage = buf.readShort()
    val paddockList = {
      val builder = collection.immutable.Seq.newBuilder[PaddockInformationsForSell.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += PaddockInformationsForSell.deserialize(buf)
      }
      builder.result()
    }

    create(pageIndex, totalPage, paddockList)
  }

}

object PaddockToSellListMessage extends PaddockToSellListMessage {
  case class MessageImpl(pageIndex: Short, totalPage: Short, paddockList: Seq[PaddockInformationsForSell.TypeImpl]) extends Message
  type Target = MessageImpl

  override def create(pageIndex: Short, totalPage: Short, paddockList: Seq[PaddockInformationsForSell.TypeImpl]) = MessageImpl(pageIndex, totalPage, paddockList)

  def apply(pageIndex: Short, totalPage: Short, paddockList: Seq[PaddockInformationsForSell.TypeImpl]) = create(pageIndex, totalPage, paddockList)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/PaddockToSellListMessage.java


// Generated on 05/08/2013 19:37:48
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PaddockToSellListMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6138;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short pageIndex;
    public short totalPage;
    public PaddockInformationsForSell[] paddockList;
    
    public PaddockToSellListMessage() { }
    
    public PaddockToSellListMessage(short pageIndex, short totalPage, PaddockInformationsForSell[] paddockList) {
        this.pageIndex = pageIndex;
        this.totalPage = totalPage;
        this.paddockList = paddockList;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(pageIndex);
        buf.writeShort(totalPage);
        buf.writeUShort(paddockList.length);
        for (PaddockInformationsForSell entry : paddockList) {
            entry.serialize(buf);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        pageIndex = buf.readShort();
        if (pageIndex < 0)
            throw new RuntimeException("Forbidden value on pageIndex = " + pageIndex + ", it doesn't respect the following condition : pageIndex < 0");
        totalPage = buf.readShort();
        if (totalPage < 0)
            throw new RuntimeException("Forbidden value on totalPage = " + totalPage + ", it doesn't respect the following condition : totalPage < 0");
        int limit = buf.readUShort();
        paddockList = new PaddockInformationsForSell[limit];
        for (int i = 0; i < limit; i++) {
            paddockList[i] = new PaddockInformationsForSell();
            paddockList[i].deserialize(buf);
        }
    }
    
}
